snyk_scm_refresh.py

```diff
@@ -80,29 +78,133 @@ def delete_stale_manifests(snyk_repo_projects):
             if not dry_run:
                 time.sleep(2)
 
+def delete_renamed_repo_projects(import_status_checks):
+    """
+    Check status of pending import jobs
+    up to PENDING_REMOVAL_MAX_CHECKS times,
+    waiting PENDING_REMOVAL_CHECK_INTERVAL seconds between checks
+    """
+    check_count = 0
+
+    print("Checking on renamed project imports before removing %d old ones."
+          % len(import_status_checks))
+    print(
+        "Polling results for up to %d mintues..." % (
+            (PENDING_REMOVAL_MAX_CHECKS * PENDING_REMOVAL_CHECK_INTERVAL)/60))
+    while check_count < PENDING_REMOVAL_MAX_CHECKS:
+        # list to track which checks to remove as we go
+        checks_to_pop = []
+        # get unique jobs from import results
+        unique_import_jobs = []
+        for import_status_check in import_status_checks:
+            if import_status_check["job_id"] not in {u["job_id"] for u in unique_import_jobs}:
+                unique_import_jobs.append(
+                    {
+                        "job_id": import_status_check["job_id"],
+                        "import_status_url": import_status_check["import_status_url"],
+                        "org_id": import_status_check["org_id"]
+                    }
+                )
 
-def delete_renamed_repo_manifests(snyk_repo_projects):
-    """Delete Snyk projects (manifests) that exist under their old repo name"""
-    for snyk_repo_project in snyk_repo_projects:
-        delete_snyk_project(snyk_repo_project["id"], snyk_repo_project["org_id"])
-        RENAMED_MANIFESTS_DELETED_FILE.write(
-            "%s,%s,%s\n"
-            % (
-                snyk_repo_project["name"],
-                snyk_repo_project["id"],
-                snyk_repo_project["org_id"],
+        # for each unique import job
+        # 1. check the status of each nested log entry (repo import), and
+        # 2. delete the corresponding stale projects under the old repo name
+
+        for import_job in unique_import_jobs:
+            import_status = get_import_status(
+                import_job["import_status_url"], import_job["org_id"]
+            )
+            for import_status_log in import_status["logs"]:
+                check_index = 0
+                for import_status_check in import_status_checks:
+                    import_status_check_full_name = \
```

Comment:
It doesn't really matter but if we specify that Python > 3.6 we can use f strings, e.g. :

`f"{import_status_check["owner"]}/{import_status_check["name"]}"`